src/analysis/retail/priest/holy/modules/core/HolyWordCDR.tsx

```diff
@@ -0,0 +1,206 @@
+import Analyzer from 'parser/core/Analyzer';
+import { Options } from 'parser/core/Module';
+import Combatants from 'parser/shared/modules/Combatants';
+
+import { TALENTS_PRIEST } from 'common/TALENTS';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import { TIERS } from 'game/TIERS';
+import { CastEvent } from 'parser/core/Events';
+import TALENTS from 'common/TALENTS/priest';
+import {
+  APOTH_MULTIPIER,
+  baseHolyWordCDR,
+  chastiseHWCDR,
+  energyCycleCDR,
+  LIGHT_OF_THE_NAARU_REDUCTION_PER_RANK,
+  salvationHWCDR,
+  sanctifyHWCDR,
+  serenityHWCDR,
+  TWW_TIER1_2PC_CDR,
+} from '../../constants';
+
+/**
+ * This module drives all holy word CDR evaluations, use as a dependency and call one of the public handlers
+ * or make a new one
+ */
+
+class HolyWordCDR extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    spellUsable: SpellUsable,
+  };
+
+  protected combatants!: Combatants;
+  protected spellUsable!: SpellUsable;
+
+  private baseHolyWordCDR = 1;
+  private baseVohMult = 0;
+  private lotnMult = 1;
+  private twwS1TierMult = 1;
+
+  constructor(options: Options) {
+    super(options);
+
+    // The below 3 if statements scale the CDR base on talents/gear
+    if (this.selectedCombatant.hasTalent(TALENTS_PRIEST.LIGHT_OF_THE_NAARU_TALENT)) {
+      this.lotnMult =
+        this.selectedCombatant.getTalentRank(TALENTS_PRIEST.LIGHT_OF_THE_NAARU_TALENT) *
+          LIGHT_OF_THE_NAARU_REDUCTION_PER_RANK +
+        1;
+    }
+
+    if (this.selectedCombatant.has2PieceByTier(TIERS.TWW1)) {
+      this.twwS1TierMult = TWW_TIER1_2PC_CDR + 1;
+    }
+
+    if (this.selectedCombatant.hasTalent(TALENTS_PRIEST.VOICE_OF_HARMONY_TALENT)) {
+      this.baseVohMult =
+        0.5 * this.selectedCombatant.getTalentRank(TALENTS_PRIEST.VOICE_OF_HARMONY_TALENT);
+    }
+
+    this.baseHolyWordCDR = this.lotnMult * this.twwS1TierMult;
+  }
+
+  // example of single holy word wrapper
+
+  /**
+  public handleSerenityCDR(event: CastEvent): hwCDRBreakdown | undefined {
+    //filter
+    if (!serenityHWCDR.has(event.ability.guid)) {
+      return;
+    }
+    return this.handleCDR(
+      event,
+      serenityHWCDR.get(event.ability.guid),
+      TALENTS.HOLY_WORD_SERENITY_TALENT.id,
+    );
+  }
+    */
```

Comment:
This comment is unnecessary, those interested can just find where this function is called instead
```suggestion

```